### PR TITLE
Support astral ID_Continue characters in capture group names

### LINKED ISSUE #90: Does not support astral ID_Continue characters in capture group names
`/(?<$𐒤>a)/u` should parse according to [test262](https://github.com/tc39/test262/blob/master/test/built-ins/RegExp/named-groups/unicode-property-names.js#L16).

### PR DESCRIPTION
**Pull request by adrianheine**
Closes #90.

### CODE REVIEW

File: tools/generate-identifier-regex.js [hunk 1]
```diff
@@ -3,49 +3,45 @@
...
-var Nd = get('categories/Nd');
-var Pc = get('categories/Pc');
+// Get the Unicode properties needed to construct the ES6 regex.
+var ID_Start = get('Binary_Property/ID_Start');
+var ID_Continue = get('Binary_Property/ID_Continue');
+var Other_ID_Start = get('Binary_Property/Other_ID_Start');
 
-var generateES5Regex = function() { // ES 5.1
-  // http://mathiasbynens.be/notes/javascript-identifiers#valid-identifier-names
+var generateES6Regex = function() {
```
Comment (mathiasbynens) [inline tools/generate-identifier-regex.js]: Let’s drop the version number altogether. As of ES2015 we decided the spec would always follow the latest available Unicode version, and the ES grammar itself is unlikely to change again.

File: tools/generate-identifier-regex.js [hunk 2]
```diff
@@ -3,49 +3,45 @@
...
     'NonAsciiIdentifierStart': identifierStart.toString(),
     'NonAsciiIdentifierPartOnly': identifierPartOnly.toString()
   };
 };
 
-var result = generateES5Regex();
+var result = generateES6Regex();
 console.log(
-  '// ECMAScript 5.1/Unicode v%s NonAsciiIdentifierStart:\n\n%s\n',
+  '// ECMAScript 6.0/Unicode v%s NonAsciiIdentifierStart:\n\n%s\n',
```
Comment (mathiasbynens) [inline tools/generate-identifier-regex.js]: Here too

File: test/test-data-named-groups-unicode.json [hunk 3]
```diff
@@ -62,5 +62,27 @@
...
+    "raw": "(?<$𐒤>a)",
+    "name": {
+      "type": "identifier",
+      "value": "$𐒤",
+      "range": [
+        3,
+        6
+      ],
+      "raw": "$𐒤"
+    }
```
Comment (mathiasbynens) [inline test/test-data-named-groups-unicode.json:86]: Maybe add a few more tests? Here are some more examples of differences between ES5 and ES2015+ identifiers: https://mathiasbynens.be/notes/javascript-identifiers-es6

**Review by mathiasbynens — APPROVED** [review]
LGTM % nits

File: test/test-data-named-groups-unicode.json (hunk 3 above)
Comment (adrianheine) [reply test/test-data-named-groups-unicode.json:86]: I added another test for two of these differences.

**Comment by jviereck** [comment]
@adrianheine, thanks for providing this addressing the review comments of @mathiasbynens!

**Comment by jviereck** [comment]
@adrianheine, do you have a patch coming up soon like with the last version? If not, I would release a new version, otherwise consider waiting for a new PR to land.

**Comment by adrianheine** [comment]
I think that's it for now :)

**Comment by jviereck** [comment]
Published v0.6.0 to npm and added a new release: https://github.com/jviereck/regjsparser/releases/tag/v0.6.0.

**Pull request merged**